frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.tsx

```diff
@@ -124,10 +157,13 @@ export class TagEditorParam extends Component {
     }
   }
 
-  setDimension(fieldId) {
+  setDimension(fieldId: FieldId) {
     const { tag, setTemplateTag, metadata } = this.props;
-    const dimension = ["field", fieldId, null];
-    if (!_.isEqual(tag.dimension !== dimension)) {
+
+    // TODO Fix raw MBQL usage
+    const dimension: DimensionReference = ["field", fieldId, null];
+
+    if (!_.isEqual(tag.dimension, dimension)) {
```

Comment:
```js
_.isEqual(tag.dimension !== dimension)
```

This doesn't make any sense, so swapping it with dimension comparison 🫣

Reply:
😫